modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,376 @@
+<?php
+/**
+ * Module Name: Dominant Color
+ * Description: Adds support to store dominant-color for an image and create a placeholder background with that color.
+ * Experimental: Yes
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+/**
+ * Add the dominant color metadata to the attachment.
+ *
+ * @since n.e.x.t
+ *
+ * @param array $metadata The attachment metadata.
+ * @param int   $attachment_id The attachment ID.
+ *
+ * @return array $metadata
+ */
+function dominant_color_metadata( $metadata, $attachment_id ) {
+	if ( ! wp_attachment_is_image( $attachment_id ) ) {
+
+		return $metadata;
+	}
+
+	$dominant_color = dominant_color_get( $attachment_id );
+
+	if ( ! empty( $dominant_color ) ) {
+		$metadata['dominant_color'] = $dominant_color;
+	}
+
+	$has_transparency = dominant_color_get_has_transparency( $attachment_id );
+
+	if ( ! empty( $has_transparency ) ) {
+		$metadata['has_transparency'] = $has_transparency;
+	} else {
+		$metadata['has_transparency'] = false;
+	}
+
+	return $metadata;
+}
+add_filter( 'wp_generate_attachment_metadata', 'dominant_color_metadata', 10, 2 );
+
+/**
+ * Filter various image attributes to add the dominant color to the image
+ *
+ * @since n.e.x.t
+ *
+ * @param array  $attr Attributes for the image markup.
+ * @param object $attachment Image attachment post.
+ *
+ * @return mixed
+ */
+function dominant_color_tag_add_adjust_to_image_attributes( $attr, $attachment ) {
+
+	$image_meta = wp_get_attachment_metadata( $attachment->ID );
+	if ( ! is_array( $image_meta ) ) {
+		return $attr;
+	}
+
+	$has_transparency = isset( $image_meta['has_transparency'] ) ? $image_meta['has_transparency'] : false;
+
+	$extra_class = '';
+	if ( ! isset( $attr['style'] ) ) {
+		$attr['style'] = ' --has-transparency: ' . $has_transparency . '; ';
+	} else {
+		$attr['style'] .= ' --has-transparency: ' . $has_transparency . '; ';
+		$extra_class    = ' has-transparency ';
+	}
+
+	if ( isset( $image_meta['dominant_color'] ) ) {
+		$dominant_color = $image_meta['dominant_color'];
+	} else {
+		/**
+		 * Filters the default color to use when no dominant color is found.
+		 *
+		 * @since n.e.x.t
+		 *
+		 * @param string $default_color The default color 'cccccc'.
+		 */
+		$dominant_color = apply_filters( 'dominant_color_default_color', 'cccccc' );
+	}
+
+	if ( ! empty( $dominant_color ) ) {
```

Comment:
Based on your above comment, we shouldn't set a background color on a transparent image. So why not include that condition here? Let's only set the `data-dominant-color` and related `style` tweaks when the image for sure is not transparent (i.e. when `$image_meta['has_transparency']` is set _and_ `false`).

Reply:
I agree here, the background attribute should only be set if value exists in post meta and if the image not transparent. I think their would be unwanted side effects otherwise.